.github/scripts/main/main.sh

```diff
@@ -52,6 +52,48 @@ make install
 export PATH="$PREFIX/bin:$PATH"
 opam --version
 
+if [[ $OPAM_DOC -eq 1 ]]; then
+  make -C doc html man-html pages
+
+  if [ "$GITHUB_EVENT_NAME" = "pull_request" ]; then
+    . .github/scripts/common/hygiene-preamble.sh
+    diff="git diff $BASE_REF_SHA..$PR_REF_SHA"
+    set +e
+    files="$($diff --name-only --diff-filter=A -- src/**/*.mli)"
+    set -e
+    if [ -n "$files" ]; then
+      echo '::group::new module(s) added - check index updated too'
+      if $diff --name-only --exit-code -- doc/index.html ; then
+        echo '::error new module(s) added but doc/index.html not updated'
+        echo "$files"
+        exit 3
+      fi
+      echo '::endgroup::new module added - checking it'
+    else
+      echo 'No new modules added'
+    fi
+  fi
+
+  htmlfiles=$(ls doc/pages/*.md | sed -e 's/\.md/.html/')
+  manfiles=$(opam --help=groff | grep '^\\fB\w' | sed 's/\\fB\(.*\)\\fR .*/doc\/man-html\/opam-\1.html/' | grep -v 'opam-opam')
```

Comment:
Just taking inspiration from `doc/man/dune`, there's a slightly simpler way:
```suggestion
  manfiles=$(opam help topics | sed -e 's|.*|doc/man-html/opam-&.html|')
```
Could also add `opam admin help topics` in the same way:
```suggestion
  manfiles=$(opam help topics | sed -e 's|.*|doc/man-html/opam-&.html|')
  manfiles="$manfiles $(opam admin help topics | sed -e 's|.*|doc/man-html/opam-admin-&.html|')"
```